### PR TITLE
Sprint10 156 arbitrary content in tl mega menu

### LINKED ISSUE #156: The Mega Menu should provide a tl-mega-menu-content element
The tl-mega-menu-content element should allow for the inclusion of arbitrary HTML and should be accepted as either a sibling to the tl-mega-menu-section, or as a child to the tl-mega-menu-secion.

### PR DESCRIPTION
**Pull request by rmathew1011**
Resolves #156 The Mega Menu should provide a tl-mega-menu-content element.

### CODE REVIEW

File: src/app/tl-mega-menu/tl-mega-menu-section/tl-mega-menu-section.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <p class="section-title">{{sectionTitle}}</p>
 <wvr-nav-list-element vertical="true">
-  <ng-content select="tl-nav-li" ngProjectAs="wvr-nav-li"></ng-content>
-</wvr-nav-list-element>
\ No newline at end of file
+  <ng-content select="tl-nav-li, tl-mega-menu-custom, tl-it-works, h3, p" ngProjectAs="wvr-nav-li"></ng-content>
```
Comment (jeremythuff) [inline src/app/tl-mega-menu/tl-mega-menu-section/tl-mega-menu-section.component.html:3]: This should only allow `tl-nav-li` and `tl-mega-menu-custom`

File: src/app/tl-mega-menu/tl-mega-menu.component.scss [hunk 2]
```diff
@@ -2,6 +2,23 @@
 @import "../shared/styles/tl-variables.scss";
 
 :host {
+  .tl-mega-menu-custom-top {
+    width: calc(var(--wvr-dropdown-menu-width) + 6px);
+    margin-left: -19px;
```
Comment (jeremythuff) [inline src/app/tl-mega-menu/tl-mega-menu.component.scss:7]: Is margin the right choice here? Should we be using padding?

File: src/index.html [hunk 3]
```diff
@@ -220,27 +220,25 @@
...
+                  hidden-in-mobile="true">
+      <tl-mega-menu-custom top-content>
+          <div class="alert alert-primary" role="alert">
+            <wvr-text value="Primary Alert Message (top-content)Primary Alert Message (top-content)
+              Primary Alert Message (top-content)Primary Alert Message (top-content)Primary Alert Message (top-content)
+              Primary Alert Message (top-content)Primary Alert Message (top-content)Primary Alert Message (top-content)">
+            </wvr-text>
+          </div>
+      </tl-mega-menu-custom>
+      <tl-mega-menu-custom top-content>
```
Comment (jeremythuff) [inline src/index.html:237]: This should be included only as a child of the `tl-mega-menu-custom`

File: src/index.html [hunk 4]
```diff
@@ -220,27 +220,25 @@
...
-          <a href="/services/option2">
-            <wvr-text value="Option 2"></wvr-text>
-          </a>
-        </tl-nav-li>
-        <tl-nav-li >
-          <a href="/services/option3">
-            <wvr-text value="Option 3"></wvr-text>
-          </a>
-        </tl-nav-li>
+        <tl-it-works text="Section 1 using jumbotron"></tl-it-works>
```
Comment (jeremythuff) [inline src/index.html:241]: This should also be included only as a child of a `tl-mega-menu-custom` element

File: src/index.html [hunk 5]
```diff
@@ -260,21 +258,8 @@
...
-          <a href="/instructional/option2">
-            <wvr-text value="Option 2"></wvr-text>
-          </a>
-        </tl-nav-li>
-        <tl-nav-li >
-          <a href="/instructional/option3">
-            <wvr-text value="Option 3"></wvr-text>
-          </a>
-        </tl-nav-li>
+        <h3><wvr-text value="Custom Content Title"></wvr-text></h3>
```
Comment (jeremythuff) [inline src/index.html:261]: These should be children of `tl-mega-menu-custom`

**Review by jeremythuff — CHANGES_REQUESTED** [review]

File: src/index.html (hunk 3 above)
Comment (rmathew1011) [reply src/index.html:237]: Added as child of tl-mega-menu-custom

File: src/index.html (hunk 4 above)
Comment (rmathew1011) [reply src/index.html:241]: Added as a child of a tl-mega-menu-custom element

File: src/index.html (hunk 3 above)
Comment (rmathew1011) [reply src/index.html:237]: Was intended to added a tl-mega-menu-custom as a sibling element to tl-mega-menu and also as a child element.

File: src/app/tl-mega-menu/tl-mega-menu.component.scss (hunk 2 above)
Comment (rmathew1011) [reply src/app/tl-mega-menu/tl-mega-menu.component.scss:7]: margin rule enabled the gap on the left to be occupied by the top-content

File: src/app/tl-mega-menu/tl-mega-menu.component.scss [hunk 6]
```diff
@@ -2,6 +2,23 @@
 @import "../shared/styles/tl-variables.scss";
 
 :host {
+  .tl-mega-menu-custom-top {
+    width: calc(var(--wvr-dropdown-menu-width) + 6px);
```
Comment (jeremythuff) [inline src/app/tl-mega-menu/tl-mega-menu.component.scss:6]: Lets try and set the drop down menu padding to 0, and to add padding to the .mega-menu-sections div.

**Review by jeremythuff — CHANGES_REQUESTED** [review]

**Review by jeremythuff — APPROVED** [review]

**Pull request merged**